main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintSolver.scala

```diff
@@ -108,11 +108,11 @@ object ConstraintSolver {
           // If there are any constraints we could not resolve, then we report an error.
           // TODO ASSOC-TYPES here we only consider the first error
           getFirstError(deferred, renv) match {
```

Comment:
I can now change this to report all errors if we want to?

Reply:
I think we should. Agree @mlutze ?